bigtable/client/internal/throw_delegate.cc

```diff
@@ -16,55 +16,9 @@
 #include "bigtable/client/grpc_error.h"
 #include <sstream>
 
-namespace {
-template <typename Exception>
-[[noreturn]] void RaiseException(char const* msg) {
-#ifdef GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
-  throw Exception(msg);
-#else
-  std::cerr << "Aborting because exceptions are disabled: " << msg << std::endl;
-  // TODO(#327) - make the call to std::abort() configurable.
-  std::abort();
-#endif  // GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
-}
-}
-
 namespace bigtable {
 inline namespace BIGTABLE_CLIENT_NS {
 namespace internal {
-
-[[noreturn]] void RaiseInvalidArgument(char const* msg) {
-  RaiseException<std::invalid_argument>(msg);
-}
-
-[[noreturn]] void RaiseInvalidArgument(std::string const& msg) {
-  RaiseException<std::invalid_argument>(msg.c_str());
-}
-
-[[noreturn]] void RaiseRangeError(char const* msg) {
-  RaiseException<std::range_error>(msg);
-}
-
-[[noreturn]] void RaiseRangeError(std::string const& msg) {
-  RaiseException<std::range_error>(msg.c_str());
-}
-
-[[noreturn]] void RaiseRuntimeError(char const* msg) {
-  RaiseException<std::runtime_error>(msg);
-}
-
-[[noreturn]] void RaiseRuntimeError(std::string const& msg) {
-  RaiseException<std::runtime_error>(msg.c_str());
-}
-
-[[noreturn]] void RaiseLogicError(char const* msg) {
-  RaiseException<std::logic_error>(msg);
-}
-
-[[noreturn]] void RaiseLogicError(std::string const& msg) {
-  RaiseException<std::logic_error>(msg.c_str());
-}
-
 [[noreturn]] void RaiseRpcError(grpc::Status const& status, char const* msg) {
 #ifdef GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
   throw bigtable::GRpcError(msg, status);
```

Comment:
It doesn't seem like there's any reason to keep the grpc_error code in bigtable/client/, but it's not hurting anything at the moment either.  It just puts off the need to refactor it out into the common namespace so it can be included (or not) by each client... but that's not important unless/until we add a client for some other service that utilzes grpc.

If you think that's worth doing at some point, I'll open an issue.

Reply:
Moving it now would require making the common library depend on gRPC, and that means the GCS C++ client would depend on gRPC, but it does not need to.  We could prevent that with some clever split of the common library, but I rather think about that clever split later, when we need to.